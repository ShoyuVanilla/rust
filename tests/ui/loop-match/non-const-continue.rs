// Test that `#[const_continue]` attribute can only be placed on constants.

#![allow(incomplete_features)]
#![feature(loop_match)]
#![crate_type = "lib"]

fn test(mut state: u8) {
    #[loop_match]
    loop {
        state = 'blk: {
            match state {
                0 => {
                    #[const_continue]
                    break 'blk state;
                    //~^ ERROR the value of this `#[const_continue]` is not a constant
                }

                _ => unreachable!(),
            }
        }
    }
}
