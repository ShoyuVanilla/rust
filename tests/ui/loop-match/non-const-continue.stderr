error: the value of this `#[const_continue]` is not a constant
  --> $DIR/non-const-continue.rs:14:32
   |
LL |                     break 'blk state;
   |                                ^^^^^ this value is not a constant
   |
   = note: the value must be a literal or a monomorphic const

error: aborting due to 1 previous error

